scripts/west_commands/runners/stm32cubeprogrammer.py

```diff
@@ -0,0 +1,185 @@
+# Copyright (c) 2020 Teslabs Engineering S.L.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+"""Runner for flashing with STM32CubeProgrammer CLI, the official programming
+   utility from ST Microelectronics.
+"""
+
+from pathlib import Path
+import os
+import shlex
+import sys
+from typing import TYPE_CHECKING, List, Optional, ClassVar, Dict
+
+from runners.core import ZephyrBinaryRunner, RunnerCaps
+
+if TYPE_CHECKING:
+    import argparse
+
+    from runners.core import RunnerConfig
+
+
+class STM32CubeProgrammerBinaryRunner(ZephyrBinaryRunner):
+    """Runner front-end for STM32CubeProgrammer CLI."""
+
+    _RESET_MODES: ClassVar[Dict[str, str]] = {
+        "sw": "SWrst",
+        "hw": "HWrst",
+        "core": "Crst",
+    }
+    """Reset mode argument mappings."""
+
+    def __init__(
+        self,
+        cfg: "RunnerConfig",
+        port: str,
+        frequency: Optional[int],
+        reset_mode: Optional[str],
+        extra_connect_opts: Optional[str],
+        prefix: Optional[Path],
+        use_elf: bool,
+        erase: bool,
+        extra_opts: List[str],
+    ) -> None:
+        super().__init__(cfg)
+
+        self._port = port
+        self._frequency = frequency
+        self._reset_mode = reset_mode
+        self._extra_connect_opts = extra_connect_opts
+        self._prefix = prefix
+        self._use_elf = use_elf
+        self._erase = erase
+
+        self._extra_opts = []
+        for opts in [shlex.split(opt) for opt in extra_opts]:
+            self._extra_opts += opts
+
+        self._stm32_programmer_cli = self._get_stm32cubeprogrammer_path()
+
+        # add required library loader path to the environment (Linux only)
+        if sys.platform == "linux":
+            os.environ["LD_LIBRARY_PATH"] = str(
```

Comment:
@mbolivar-nordic would be nice to allow args/kwargs on `check_call`, so that environment can be set using `env` on that call.

Reply:
> @mbolivar-nordic would be nice to allow args/kwargs on `check_call`, so that environment can be set using `env` on that call.

Sure, feel free to add this in a separate commit in this PR.